datadog fails to compile tests

```
/var/stackage/work/unpack-dir/unpacked/datadog-0.2.4.0-cbe46ca40b0d9697945a69be51ac5b1d4c2b28677180f273ed9df2fcf411790b/test/Test/Network/Datadog/StatsD.hs:16:5: error:
    Module ‘Network.Socket’ does not export ‘recvFrom’
   |
16 |   , recvFrom
   |     ^^^^^^^^

--  End of log file: /var/stackage/work/unpack-dir/.stack-work/logs/datadog-0.2.4.0.log
```

CC @dfithian

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -5123,7 +5123,6 @@ skipped-tests:
     - system-fileio # ansi-terminal-0.8 via chell
     - system-filepath # ansi-terminal-0.8 via chell
     - buffer-builder # per HTF per cpphs per polyparse (ghc 8.6 failure)
-    - datadog # https://github.com/commercialhaskell/stackage/issues/5297
 
     # Blocked by stackage upper bounds. These can be re-enabled once
     # the relevant stackage upper bound is lifted.

